pkg/gohai/network/network.go

```diff
@@ -6,29 +6,292 @@
 // Package network regroups collecting information about the network interfaces
 package network
 
-func getNetworkInfo() (networkInfo map[string]interface{}, err error) {
-	networkInfo = make(map[string]interface{})
+import (
+	"errors"
+	"fmt"
+	"net"
 
+	"github.com/DataDog/datadog-agent/pkg/gohai/utils"
+)
+
+// ErrAddressNotFound means no such address could be found
+var ErrAddressNotFound = errors.New("address not found")
+
+// Interface holds information about a specific interface
+type Interface struct {
+	// Name is the name of the interface
+	Name string `json:"name"`
+	// IPv4Network is the ipv4 address for the host
+	IPv4Network utils.Value[string] `json:"ipv4-network"`
+	// IPv6Network is the ipv6 address for the host
+	IPv6Network utils.Value[string] `json:"ipv6-network"`
+	// MacAddress is the mac address for the host
+	MacAddress utils.Value[string] `json:"macaddress"`
+	// IPv4 is the list of IPv4 addresses for the interface
+	IPv4 []string `json:"ipv4"`
+	// IPv4 is the list of IPv6 addresses for the interface
+	IPv6 []string `json:"ipv6"`
+}
+
+// Info holds network metadata about the host
+type Info struct {
+	// Interfaces is the list of interfaces which are up
+	Interfaces []Interface `json:"interfaces"`
+	// MacAddress is a mac address of the host
+	MacAddress string `json:"macaddress"`
+	// IPAddress is an IPv4 address of the host
+	IPAddress string `json:"ipaddress"`
+	// IPAddressV6 is an IPv6 address of the host
+	IPAddressV6 utils.Value[string] `json:"ipaddressv6"`
+}
+
+// CollectInfo collects the network information.
+func CollectInfo() (*Info, error) {
+	info := &Info{}
+	err := fillNetworkInfo(info)
```

Comment:
Why does `fillNetworkInfo` doesnt create the `Info` struct itself (we could rename it `getNetworkInfo`) ?

Reply:
I initially didn't go with `getNetworkInfo` because it wouldn't return the full network info (it wouldn't return the interface list).
But it doesn't change much I'll rename it